src/java/org/apache/cassandra/service/opstate/KeyspaceTableOpStatePersister.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.opstate;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.codehaus.jackson.map.ObjectMapper;
+
+public class KeyspaceTableOpStatePersister
+{
+    private static final Logger log = LoggerFactory.getLogger(KeyspaceTableOpStatePersister.class);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private File persistentFile;
+    private final String persistentFileLocation;
+
+    public KeyspaceTableOpStatePersister(String persistentFileLocation)
+    {
+        this.persistentFileLocation = persistentFileLocation;
+        this.persistentFile = getOrMaybeCreateStateFile();
+    }
+
+    public Map<KeyspaceTableKey, Instant> readStateFromPersistentLocation()
+    {
+        File persistentStateFile = getOrMaybeCreateStateFile();
+        if (persistentStateFile == null)
+            return new HashMap<>();
+
+        try
+        {
+            return readStateFromFile(persistentStateFile);
+        }
+        catch (IOException e)
+        {
+            return new HashMap<>();
+        }
+    }
+
+    public boolean updateStateInPersistentLocation(Map<KeyspaceTableKey, Instant> updatedEntries)
+    {
+        File persistentStateFile = getOrMaybeCreateStateFile();
+        if (persistentStateFile == null)
+            return false;
+        try
+        {
+            Map<KeyspaceTableKey, Instant> persistentEntries = readStateFromFile(persistentStateFile);
+
+            // We assume inputted entries will always be more up to date compared to the ones from the file
```

Comment:
Do you think that is an acceptable assumption? Or should we explicitly check (and potentially throw)?

Reply:
Unless I am mistaken somehow, this is what it is currently doing by using the `readStateFromFile` function which throws if it cannot read?